File: src/events/PallyConEventType.ts
/// Event type of PallyCon SDK Event
export enum PallyConEventType {
    /// The download completed
    complete = 'complete',
    /// The download paused
    pause = 'pause',
    /// The download is removed
    remove = 'remove',
    /// The download is stop
    stop = 'stop',
    /// The download is start
    download = 'download',
    /// Error when the content information to be downloaded is incorrect
    /// It mainly occurs when the value of the function parameter [PallyConContentConfiguration] is incorrect.
    contentDataError = 'contentDataError',
    /// License error
    drmError = 'drmError',
    /// Server error when issuing license
    licenseServerError = 'licenseServerError',
    /// Error during download
    downloadError = 'downloadError',
    /// Error when there is no network connection
    networkConnectedError = 'networkConnectedError',
    /// Error that occurs when the time is forcibly manipulated on an Android device
    detectedDeviceTimeModifiedError = 'detectedDeviceTimeModifiedError',
    /// Error that occurs when migrating from SDK
    migrationError = 'migrationError',
    /// Error that occurs when licenseCipher from SDK
    licenseCipherError = 'licenseCipherError',
    /// Unknown error type
    unknownError = 'unknownError',
    /// Download progress data
    progress = 'progress'
}

File: ios/models/PallyConEventType.swift
import Foundation

enum PallyConEventType {
    case complete
    case download
    case pause
    case remove
    case stop
    case contentDataError
    case drmError
    case licenseServerError
    case downloadError
    case networkConnectedError
    case detectedDeviceTimeModifiedError
    case migrationError
    case licenseCipherError
    case unknownError
    case progress
}

extension PallyConEventType {
    var name: String {
        switch self {
        case .complete: return "complete"
        case .download: return "download"
        case .pause: return "pause"
        case .remove: return "remove"
        case .stop: return "stop"
        case .contentDataError: return "contentDataError"
        case .drmError: return "drmError"
        case .licenseServerError: return "licenseServerError"
        case .downloadError: return "downloadError"
        case .networkConnectedError: return "networkConnectedError"
        case .detectedDeviceTimeModifiedError: return "detectedDeviceTimeModifiedError"
        case .migrationError: return "migrationError"
        case .licenseCipherError: return "licenseCipherError"
        case .unknownError: return "unknownError"
        case .progress: return "progress"
        }
    }
}
